### ISSUE TITLE
invalid conversion from 'const wf_geometry* {aka const wlr_box*}' to 'wlr_box*'

### ISSUE THREAD

**Issue by colemickens** [WayfireWM/wayfire#112]
I'm getting build failures, possibly due to a newer wlroots:

`../src/util.cpp:60:30: error: invalid conversion from 'const wf_geometry* {aka const wlr_box*}' to 'wlr_box*' [-fpermissive]`

more log:
```
[25/148] Generating 'src/src@@wayfire@exe/xdg-shell-unstable-v6-protocol.h'.
[26/148] Generating 'src/src@@wayfire@exe/xdg-output-unstable-v1-protocol.h'.
[27/148] Generating 'src/src@@wayfire@exe/wayfire-shell-protocol.h'.
[28/148] Generating 'src/src@@wayfire@exe/wlr-layer-shell-unstable-v1-protocol.h'.
[29/148] Generating 'src/src@@wayfire@exe/xdg-shell-unstable-v6-client-protocol.h'.
[30/148] Generating 'src/src@@wayfire@exe/xdg-output-unstable-v1-client-protocol.h'.
[31/148] Generating 'src/src@@wayfire@exe/wayfire-shell-client-protocol.h'.
[32/148] Compiling C object 'proto/proto@@wl_protos@sta/meson-generated_xdg-output-unstable-v1-protocol.c.o'.
[33/148] Compiling C object 'proto/proto@@wl_protos@sta/meson-generated_wayfire-shell-protocol.c.o'.
[34/148] Compiling C object 'proto/proto@@wl_protos@sta/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o'.
[35/148] Linking static target proto/libwl_protos.a.
[36/148] Compiling C++ object 'src/src@@wayfire@exe/util.cpp.o'.
FAILED: src/src@@wayfire@exe/util.cpp.o
/nix/store/klci955kxil7q32ggran6lnhpnkk8yjz-gcc-wrapper-7.3.0/bin/c++ -Isrc/src@@wayfire@exe -Isrc -I../src -I. -I../ -I../src/api -I/nix/store/9a075za4462rbn75hkzsdlac77af7w5z-wayland-1.16.0/include -I/nix/store/pfj0fwxiv5zpzyr8r7m5333lllv8sxjz-wlroots-84c904752f10e24f052beaf12eb3cc529f1bb38d/include -I/nix/store/0mxbjs53ylb32m9y1vvkkf246yknapgy-libxkbcommon-0.8.0-dev/include -I/nix/store/r0pz7lasjz6imwj1n7j2r7nw53syijy4-libinput-1.12.3-dev/include -I/nix/store/snvbfmvzy0cq8r1zamnsi34nwrfkm839-pixman-0.36.0/include/pixman-1 -I/nix/store/fq401fq2fj6f0wx40ih9xs3w47s2hdj6-libdrm-2.4.96-dev/include -I/nix/store/fq401fq2fj6f0wx40ih9xs3w47s2hdj6-libdrm-2.4.96-dev/include/libdrm -I/nix/store/nw0jglav42l27s0rcpnsdgvfxdfjkwbh-mesa-noglu-18.2.6-dev/include -I/nix/store/myd2khzpdwsjq8gpbxj13zd4iwcqd223-libevdev-1.6.0/include/libevdev-1.0/ -I/nix/store/99gyscrngpdfgflfs4m0qd1hxhxja0mx-glm-0.9.8.5/include -I/nix/store/alfs2c8j95kffvq5pcy2s50zzkg68b43-wf-config-6d3426e216ac62ffa035035f9c1bea074e184018/include/wayfire -I/nix/store/258sh40r4f2vgm3gxj4ai1dzlwiaa3sm-libjpeg-turbo-2.0.1-dev/include -I/nix/store/a50sg2xm6vflzx5kma3sih39dq9wv7z6-zlib-1.2.11-dev/include -I/nix/store/fv8b852i96pz34bp42g0gcygm3j2vipm-libpng-apng-1.6.35-dev/include/libpng16 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -std=c++11
-O3 -DWLR_USE_UNSTABLE -Wno-unused-parameter  -MD -MQ 'src/src@@wayfire@exe/util.cpp.o' -MF 'src/src@@wayfire@exe/util.cpp.o.d' -o 'src/src@@wayfire@exe/util.cpp.o' -c ../src/util.cpp
../src/util.cpp: In function 'bool operator&(const wf_geometry&, const wf_geometry&)':
../src/util.cpp:53:33: error: invalid conversion from 'const wf_geometry* {aka const wlr_box*}' to 'wlr_box*' [-fpermissive]
     return wlr_box_intersection(&r1, &r2, &result);
                                 ^~~
In file included from ../src/api/util.hpp:8:0,
                 from ../src/util.cpp:1:
/nix/store/pfj0fwxiv5zpzyr8r7m5333lllv8sxjz-wlroots-84c904752f10e24f052beaf12eb3cc529f1bb38d/include/wlr/types/wlr_box.h:24:6: note:   initializing argument 1 of 'bool wlr_box_intersection(wlr_box*, const wlr_box*, const wlr_box*)'
 bool wlr_box_intersection(struct wlr_box *dest, const struct wlr_box *box_a,
      ^~~~~~~~~~~~~~~~~~~~
../src/util.cpp: In function 'wf_geometry wf_geometry_intersection(const wf_geometry&, const wf_geometry&)':
../src/util.cpp:60:30: error: invalid conversion from 'const wf_geometry* {aka const wlr_box*}' to 'wlr_box*' [-fpermissive]
     if (wlr_box_intersection(&r1, &r2, &result))
                              ^~~
In file included from ../src/api/util.hpp:8:0,
                 from ../src/util.cpp:1:
/nix/store/pfj0fwxiv5zpzyr8r7m5333lllv8sxjz-wlroots-84c904752f10e24f052beaf12eb3cc529f1bb38d/include/wlr/types/wlr_box.h:24:6: note:   initializing argument 1 of 'bool wlr_box_intersection(wlr_box*, const wlr_box*, const wlr_box*)'
 bool wlr_box_intersection(struct wlr_box *dest, const struct wlr_box *box_a,
      ^~~~~~~~~~~~~~~~~~~~
[37/148] Compiling C++ object 'src/src@@wayfire@exe/core_opengl.cpp.o'.
FAILED: src/src@@wayfire@exe/core_opengl.cpp.o
/nix/store/klci955kxil7q32ggran6lnhpnkk8yjz-gcc-wrapper-7.3.0/bin/c++ -Isrc/src@@wayfire@exe -Isrc -I../src -I. -I../ -I../src/api -I/nix/store/9a075za4462rbn75hkzsdlac77af7w5z-wayland-1.16.0/include -I/nix/store/pfj0fwxiv5zpzyr8r7m5333lllv8sxjz-wlroots-84c904752f10e24f052beaf12eb3cc529f1bb38d/include -I/nix/store/0mxbjs53ylb32m9y1vvkkf246yknapgy-libxkbcommon-0.8.0-dev/include -I/nix/store/r0pz7lasjz6imwj1n7j2r7nw53syijy4-libinput-1.12.3-dev/include -I/nix/store/snvbfmvzy0cq8r1zamnsi34nwrfkm839-pixman-0.36.0/include/pixman-1 -I/nix/store/fq401fq2fj6f0wx40ih9xs3w47s2hdj6-libdrm-2.4.96-dev/include -I/nix/store/fq401fq2fj6f0wx40ih9xs3w47s2hdj6-libdrm-2.4.96-dev/include/libdrm -I/nix/store/nw0jglav42l27s0rcpnsdgvfxdfjkwbh-mesa-noglu-18.2.6-dev/include -I/nix/store/myd2khzpdwsjq8gpbxj13zd4iwcqd223-libevdev-1.6.0/include/libevdev-1.0/ -I/nix/store/99gyscrngpdfgflfs4m0qd1hxhxja0mx-glm-0.9.8.5/include -I/nix/store/alfs2c8j95kffvq5pcy2s50zzkg68b43-wf-config-6d3426e216ac62ffa035035f9c1bea074e184018/include/wayfire -I/nix/store/258sh40r4f2vgm3gxj4ai1dzlwiaa3sm-libjpeg-turbo-2.0.1-dev/include -I/nix/store/a50sg2xm6vflzx5kma3sih39dq9wv7z6-zlib-1.2.11-dev/include -I/nix/store/fv8b852i96pz34bp42g0gcygm3j2vipm-libpng-apng-1.6.35-dev/include/libpng16 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -std=c++11
-O3 -DWLR_USE_UNSTABLE -Wno-unused-parameter  -MD -MQ 'src/src@@wayfire@exe/core_opengl.cpp.o' -MF 'src/src@@wayfire@exe/core_opengl.cpp.o.d' -o 'src/src@@wayfire@exe/core_opengl.cpp.o' -c ../src/core/opengl.cpp
../src/core/opengl.cpp: In member function 'wlr_box wf_framebuffer::framebuffer_box_from_damage_box(wlr_box) const':
../src/core/opengl.cpp:388:62: error: cannot convert 'wl_output_transform' to 'const wlr_box*' for argument '2' to 'void wlr_box_transform(wlr_box*, const wlr_box*, wl_output_transform, int, int)'
     wlr_box_transform(&box, transform, width, height, &result);
                                                              ^
[38/148] Compiling C++ object 'src/src@@wayfire@exe/core_seat_cursor.cpp.o'.
[39/148] Compiling C++ object 'src/src@@wayfire@exe/main.cpp.o'.
../src/main.cpp: In function 'int handle_config_updated(int, uint32_t, void*)':
../src/main.cpp:57:9: warning: ignoring return value of 'ssize_t read(int, void*, size_t)', declared with attribute warn_unused_result [-Wunused-result]
     read(fd, buf, INOT_BUF_SIZE);
     ~~~~^~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
builder for '/nix/store/7j4xbw58ci3x4k8v1fkzbq1m8mzj75zc-wayfire-6a626007b395ec0ba08bef03a1a71e5761314dc2.drv' failed with exit code 1
error: build of '/nix/store/7j4xbw58ci3x4k8v1fkzbq1m8mzj75zc-wayfire-6a626007b395ec0ba08bef03a1a71e5761314dc2.drv' failed

[cole@chimera:/etc/nix/overlays/nixpkgs-wayland]$  
```

**Comment by ammen99** [comment]
Yep, there has been a breaking change in wlroots very recently. Try building with https://github.com/swaywm/wlroots/commit/36ba56140ae325ae398229a689f20ded126608bc. I plan to wait for the next wlroots release and then stick to it.

**Comment by myfreeweb** [comment]
I posted a fix for this in #121.

**Comment by myfreeweb** [comment]
This should be closed now, all that has been merged
